File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        freq_array = [0] * 26
        for char in s:
            freq_array[ord(char)-97] += 1
        for char in t:
            val = freq_array[ord(char)-97]
            if val == 0:
                return False
            freq_array[ord(char)-97] -= 1
        return True
            

        

File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        int n = s.length();
        if (n != t.length())
            return false;
        
        int[] freqArray = new int[26];
        
        for(int i = 0 ; i < n ; ++i) {
            int s_idx = s.charAt(i) - 'a';
            freqArray[s_idx] ++;
            
            int t_idx = t.charAt(i) - 'a';
            freqArray[t_idx] --;
        }
    
        for(int i = 0 ; i < 26 ; ++i) {
            if(freqArray[i] != 0)
                return false;
        }
    return true;
    }
}